File: csharp/phone-number/PhoneNumber.cs
using System.Text.RegularExpressions;

public class PhoneNumber
{
  private string number { get; set; }

  public PhoneNumber(string num)
  {
    number = FormatNumber(num);
  }

  public string Number()
  {
    return number;
  }

  public string AreaCode()
  {
    return number.Substring(0, 3);
  }

  public override string ToString()
  {
    string a = number.Substring(0, 3),
      b = number.Substring(3, 3),
      c = number.Substring(6, 4);

    return $"({a}) {b}-{c}";
  }

  private string FormatNumber(string num)
  {
    Regex pattern = new Regex(@"\W");
    num = pattern.Replace(num, "");

    if (num.Length == 10) 
      return num;
    else if (num.Length == 11 && num.StartsWith("1")) 
      return num.Substring(1);
    else 
      return "0000000000";
  }
}

File: javascript/phone-number/phone-number.js
function PhoneNumber(input) {
  this.input = input.replace(/\W/g, "");
  this.number = function() {
    if (this.input.length === 10) return this.input;
    else if (this.input.length === 11 && this.input.charAt(0) === '1') return this.input.substring(1);
    else return "0000000000";
  }
  this.areaCode = function() {
      return this.number().slice(0, 3);
  }
  this.toString = function() {
    return "(" + this.areaCode() + ") " + this.number().slice(3, 6) + "-" + this.number().slice(6);
  }
};

module.exports = PhoneNumber;
